src/awsGuard.ts

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+    EnforcementLevel,
+    Policies,
+    PolicyPack,
+    ResourceValidationPolicy,
+    StackValidationPolicy,
+} from "@pulumi/policy";
+
+import { defaultEnforcementLevel, isEnforcementLevel } from "./enforcementLevel";
+
+const defaultPolicyPackName = "pulumi-awsguard";
+
+// Internal map that holds all the registered policy factories.
+const factoryMap: Record<string, PolicyFactory<any>> = {};
+
+/**
+ * A policy pack of rules to enforce AWS best practices for security, reliability, cost, and more!
+ *
+ * Create an instance of AwsGuard without any arguments to enable all policies with their default configuration:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard();
+ * ```
+ *
+ * The above is equivalent to:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({ all: "mandatory" });
+ * ```
+ *
+ * To make all policies advisory rather than mandatory:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({ all: "advisory" });
+ * ```
+ *
+ * To make all policies advisory, but changing a couple to be mandatory:
+ *
+ * ```typescript
+ * const awsGuard = new AwsGuard({
+ *     all: "advisory",
+ *     ec2InstanceNoPublicIP: "mandatory",
+ *     elbAccessLoggingEnabled: "mandatory",
```

Comment:
I considered using the policy names themselves instead of the names like this. However, the policy names include hyphens which would necessitate specifying as strings:

```typescript
const awsGuard = new AwsGuard({
    "ec2-instance-no-public-ip": "advisory",
    "elb-logging-enabled": "advisory",
});
```

I discussed with Erin and doing it this way doesn't feel as idiomatic as is typical with TS/JS where properties are camelCased. I'd prefer to keep these camelCase for TS/JS similar to the args passed to Pulumi resources.

If we eventually add a more general (lower-level) way of configuring policy packs, which uses the policy names (with hyphens) as the keys, we could still use this more idiomatic camelCasing scheme as I've proposed here, but under the covers map the camelCase names to hyphen names.